### PR TITLE
build using go 1.18.1

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Following #2464
xref https://github.com/kubernetes/kubernetes/issues/108910

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ameukam** [comment]
Ref: https://github.com/kubernetes/release/issues/2499
cc @kubernetes/release-engineering

File: images/releng/k8s-ci-builder/variants.yaml [hunk 1]
```diff
@@ -5,11 +5,11 @@ variants:
     OS_CODENAME: 'bullseye'
```
Comment (saschagrunert) [inline images/releng/k8s-ci-builder/variants.yaml:5]: I think we should also switch the default config, right?

**Comment by cpanato** [comment]
would close this in favor or https://github.com/kubernetes/release/pull/2500

**Comment by saschagrunert** [comment]
> would close this in favor or #2500

Alright, let's go this route.

File: images/releng/k8s-ci-builder/variants.yaml (hunk 1 above)
Comment (pacoxu) [reply images/releng/k8s-ci-builder/variants.yaml:5]: 1.17.8 is using bullseye.
Should we use `buster` for 1.18.1?

**Comment by pacoxu** [comment]
fine

**Pull request closed without merging**